test/suites/edgexfoundry/config_test.go

```diff
@@ -34,3 +39,96 @@ func TestEnvConfigChangePort(t *testing.T) {
 	utils.SnapRestart(t, supportSchedulerService)
 	utils.WaitServiceOnline(t, 60, supportSchedulerServicePort)
 }
+
+func TestChangeStartupMsg_global(t *testing.T) {
+	const (
+		defaultStartupMsg = "This is the Support Scheduler Microservice"
+
+		newStartupMsg = "snap-testing-startup-message"
+		startupMsgKey = "config.service-startupmsg"
+	)
+
+	// start clean
+	utils.SnapStop(t, supportSchedulerService)
+
+	t.Cleanup(func() {
+		utils.SnapUnset(t, platformSnap, startupMsgKey)
+		utils.SnapStop(t, supportSchedulerService)
+	})
+
+	// enable new config option to avoid mixed options issue with old env option
+	utils.SnapSet(t, platformSnap, "app-options", "true")
+
+	// make sure the startupmsg is the default testing message before setting it
+	utils.SnapStart(t, supportSchedulerService)
+	require.True(t, checkStartupMsg(t, supportSchedulerService, defaultStartupMsg))
```

Comment:
The error message is hard to follow when this fails. Better add a message instead of comment in code.